files/default/ec2_dev_2_volid.py

```diff
@@ -33,20 +33,19 @@
 region = region[:-1]
 
 # Connect to AWS using boto
-conn = boto.ec2.connect_to_region(region)
+ec2 = boto3.client('ec2', region_name=region)
 
 # Poll for blockdevicemapping
-attrib = conn.get_instance_attribute(instanceId, 'blockDeviceMapping')
-devmap = attrib.get('blockDeviceMapping')
+devices = ec2.describe_instance_attribute(InstanceId=instanceId, Attribute='blockDeviceMapping').get('BlockDeviceMappings')
 x = 0
-while not devmap.has_key(dev):
+while not any(d['DeviceName'] == dev for d in devices):
```

Comment:
I realize this is confusing... the new format for BlockDeviceMappings is a list of dicts. 

In light of this I renamed `devmap` to `devices` and changed the logic to find the right item in the list.